parasolr/solr/client.py

```diff
@@ -39,33 +39,43 @@ def __repr__(self):
         return "ParasolrDict(%s)" % super(AttrDict, self).__repr__()
 
 
-class QueryResponse:
-    """Thin wrapper to give access to Solr select responses.
-
-    Args:
-        response: A Solr query response
-    """
+class BaseResponse:
+    """Base Solr response class with fields common to standard and
+    grouped results."""
 
     def __init__(self, response: Dict) -> None:
-        # cast to ParasolrDict for any dict-like object
-        response = ParasolrDict(response)
-        self.numFound = int(response.response.numFound)
-        self.start = int(response.response.start)
-        self.docs = response.response.docs
-        self.params = response.responseHeader.params
-        self.stats = response.stats if "stats" in response else {}
+        # cast to ParasolrDict for any dict-like object and store
+        self.response = ParasolrDict(response)
+        # facet counts need to be processed to convert into
+        # ordered dict, so process and store
         self.facet_counts = {}
-        if "docs" in response.response:
-            self.docs = response.response.docs
-        if "facet_counts" in response:
+        if "facet_counts" in self.response:
             self.facet_counts = self._process_facet_counts(response.facet_counts)
-        self.highlighting = response.get("highlighting", {})
-        self.expanded = response.get("expanded", {})
 
         # NOTE: To access facet_counts.facet_fields or facet_counts.facet_ranges
         # as OrderedDicts, you must use dict notation (or AttrDict *will*
         # convert).
 
+    @property
+    def params(self):
+        "parameters sent to solr in the request, as returned in response header"
+        return self.response.responseHeader.params
+
+    @property
+    def stats(self):
+        "stats portion of the response, if statics were requested"
+        return self.response.get("stats", {})
+
+    @property
+    def highlighting(self):
+        "highlighting portion of the response, if highlighting was requested"
+        return self.response.get("highlighting", {})
+
+    @property
+    def expanded(self):
+        "expanded portion of the response, if collapse/expanded results enabled"
+        return self.response.get("expanded", {})
+
```

Comment:
This is nice and extensible

Reply:
I hope so! It seemed cleaner than setting them in the constructor